src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryDate.cs

```diff
@@ -719,6 +717,54 @@ public static FormulaValue Weekday(EvalVisitor runner, EvalVisitorContext contex
             return NumberOrDecimalValue(irContext, weekday);
         }
 
+        public static FormulaValue WeekNum(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
+        {
+            var timeZoneInfo = runner.TimeZoneInfo;
+            var startOfWeek = 0d;
+
+            if (args[0] is BlankValue)
+            {
+                return NumberOrDecimalValue(irContext, 0);
+            }
+
+            var arg0 = runner.GetNormalizedDateTime(args[0]);
+            var dow = arg0.DayOfWeek;
+
+            if (args[1] is ErrorValue)
+            {
+                return args[1];
+            }
+
+            startOfWeek = GetDoubleFromFormulaValue(args[1], 1);
+
+            if (startOfWeek <= 0 || startOfWeek > 17 || (startOfWeek > 3 && startOfWeek < 11) || startOfWeek != Math.Floor(startOfWeek))
```

Comment:
If we force the maker to use the enumeration value statically, we won't need those checks

Reply:
Yes. but then we are breaking the semantics since PA allows it. Is there an easy way to handle this from PA side?

Reply:
The Weekday function handles that. See the tests, where we have two versions of tests with numbers as StartOfWeek argument, one with StronglyTypedBuiltinEnums enabled one with it disabled (the former returns errors, the latter works)